src/main/java/com/hms_networks/americas/sc/extensions/historicaldata/HistoricalDataQueueManager.java

```diff
@@ -334,87 +334,61 @@ public static synchronized ArrayList getFifoNextSpanData(
+    boolean stringHistorical = false;
+    HistoricalDataManager.exportHistoricalToFile(
+        ebdStartTime,
+        ebdEndTime,
+        ebdFileName,
+        includeTagGroupA,
+        includeTagGroupB,
+        includeTagGroupC,
+        includeTagGroupD,
+        stringHistorical);
+
+    // Parse standard EBD export call
+    queueData = HistoricalDataManager.parseHistoricalFile(ebdFileName);
+
+    // Run string EBD export call if enabled
+    if (stringHistoryEnabled) {
+      final String ebdStringFileName =
           HistoricalDataConstants.QUEUE_FILE_FOLDER
               + "/"
-              + HistoricalDataConstants.QUEUE_EBD_FILE_NAME
+              + HistoricalDataConstants.QUEUE_EBD_STRING_FILE_NAME
               + HistoricalDataConstants.QUEUE_FILE_EXTENSION;
-      boolean stringHistorical = false;
+      stringHistorical = true;
       HistoricalDataManager.exportHistoricalToFile(
           ebdStartTime,
           ebdEndTime,
-          ebdFileName,
+          ebdStringFileName,
           includeTagGroupA,
           includeTagGroupB,
           includeTagGroupC,
           includeTagGroupD,
           stringHistorical);
 
-      // Parse standard EBD export call
-      queueData = HistoricalDataManager.parseHistoricalFile(ebdFileName);
-
-      // Run string EBD export call if enabled
-      if (stringHistoryEnabled) {
-        final String ebdStringFileName =
-            HistoricalDataConstants.QUEUE_FILE_FOLDER
-                + "/"
-                + HistoricalDataConstants.QUEUE_EBD_STRING_FILE_NAME
-                + HistoricalDataConstants.QUEUE_FILE_EXTENSION;
-        stringHistorical = true;
-        HistoricalDataManager.exportHistoricalToFile(
-            ebdStartTime,
-            ebdEndTime,
-            ebdStringFileName,
-            includeTagGroupA,
-            includeTagGroupB,
-            includeTagGroupC,
-            includeTagGroupD,
-            stringHistorical);
-
-        // Parse string EBD export call and combine with standard EBD call results
-        ArrayList queueStringData = HistoricalDataManager.parseHistoricalFile(ebdStringFileName);
-        queueData.addAll(queueStringData);
-      }
-
-      // Store end time +1 ms (to prevent duplicate data)
-      final String newTimeTrackerVal = Long.toString(endTimeTrackerMsLong + 1);
-      FileAccessManager.writeStringToFile(writeFile, newTimeTrackerVal);
+      // Parse string EBD export call and combine with standard EBD call results
+      ArrayList queueStringData = HistoricalDataManager.parseHistoricalFile(ebdStringFileName);
+      queueData.addAll(queueStringData);
     }
 
+    // Store end time +1 ms (to prevent duplicate data)
+    final String newTimeTrackerVal = Long.toString(endTimeTrackerMsLong + 1);
+    FileAccessManager.writeStringToFile(writeFile, newTimeTrackerVal);
+
     isFile1CurrTimeTrackerFile = !isFile1CurrTimeTrackerFile;
 
     // Return data
     return queueData;
   }
 
-  /**
```

Comment:
Has this been tested thoroughly? I cannot remember the details, but I thought that enoughSeconds() was implemented due to a bug we saw, and it fixed that behavior. I think Oliver has taken that on. See https://github.com/hms-networks/sc-flexy-tag-data-lib/commit/ce6fe4e11e44eea67f725a86ba00d893685314bf

Reply:
I have manually tested this with no abnormal behavior.  It always returns just the header.  I did abuse it a bit too ( tried the 33rd day of January, tried the 75th second) and it stood up... it just returns the header.  
How much you want to be about there being a bug on February 29th of a leap year?